hbase-client/src/main/java/org/apache/hadoop/hbase/client/Query.java

```diff
@@ -164,7 +164,11 @@ public int getReplicaId() {
    * @param level IsolationLevel for this query
    */
   public Query setIsolationLevel(IsolationLevel level) {
-    setAttribute(ISOLATION_LEVEL, level.toBytes());
+    if (this instanceof ImmutableScan) {
```

Comment:
Why?

Reply:
Because while constructing ImmutableScan(Scan), it calls parent Scan(Scan), which calls super.setIsolationLevel() and here, it calls setAttribute() which is overridden by ImmutableScan and hence, it will fail. I know this is not good check, but we do need to directly call `super.setAttribute()`.